.github/workflows/flow-micro-benchmarks.yml

```diff
@@ -0,0 +1,122 @@
+name: Run a Micro Benchmark Flow
+
+on:
+    # Manual trigger
+    workflow_dispatch:
+      inputs:
+        profile_env:
+          type: number # for default of 0
+        rejson_branch:
+          type: string
+          default: master
+          description: 'branch to use for rejson'
+        benchmark_folder: 
+          type: string
+          default: src/redisearch_rs/trie_bencher
+          description: folder containing the benchmark to be run with `cargo bench`
+        output_folder:
+          type: string
+          default: bin/criterion/trie
+          description: folder used to store the output of criterion
+        memory_limit:
+          type: number
+          default: 12582912 # that's 12GB
+          description: Used by the memory monitoring to understand if a runner terminates because of lack of memory (leaking)
+    
+    # For calling from other workflows
+    workflow_call:
+      inputs:
+        profile_env:
+          type: number # for default of 0
+        rejson_branch:
+          type: string
+          default: master
+          description: 'branch to use for rejson'
+        benchmark_folder: 
+          type: string
+          default: src/redisearch_rs/trie_bencher
+          description: folder containing the benchmark to be run with `cargo bench`
+        output_folder:
+          type: string
+          default: bin/criterion/trie
+          description: folder used to store the output of criterion
```

Comment:
I think it's premature to try to parametrise this since different setups may be required depending on the benchmark.
I would suggest to keep it simple first, and then generalize when we have multiple benchmark suites.